src/XMakeBuildEngine/Evaluation/IntrinsicFunctions.cs

```diff
@@ -364,6 +364,17 @@ internal static bool DoesTaskHostExist(string runtime, string architecture)
             return false;
         }
 
+        internal static bool IsOSPlatform(string os)
+        {
+            switch (os.ToLower())
+            {
+                case "windows": return NativeMethodsShared.IsWindows;
+                case "osx": return NativeMethodsShared.IsOSX;
+                case "unix": return NativeMethodsShared.IsUnixLike;
+                default: return false;
```

Comment:
I think this makes sense for full-framework and Mono builds, since the CoreFX method doesn't exist there. When it does, though, should this be a passthrough to `RuntimeInformation.IsOSPlatform(OSPlatform.Create(os))`?